lib/stack_master/template_compiler.rb

```diff
@@ -1,8 +1,11 @@
 module StackMaster
   class TemplateCompiler
+    TemplateCompilationFailed = Class.new(RuntimeError)
 
     def self.compile(config, template_file_path)
       template_compiler_for_file(template_file_path, config).compile(template_file_path)
+    rescue StandardError
```

Comment:
@stevehodgkiss @nemski We don't print out the 'cause' right now. I'm wondering if we should include more than just the cause error's message in our output. May be `cause.inspect` would be better?

message:
```
"(<unknown>): mapping values are not allowed in this context at line 1 column 8"
```

Inspect:
```
#<Psych::SyntaxError: (<unknown>): mapping values are not allowed in this context at line 1 column 8>
```